File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/StorageForm.ts
import { StringEditor, EnumEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EStorageType } from "../Web/Enums.EStorageType";

export interface StorageForm {
    AccessKeyId: StringEditor;
    SecretAccessKey: StringEditor;
    Endpoint: StringEditor;
    Region: StringEditor;
    BucketName: StringEditor;
    S3Types: EnumEditor;
    IsDefault: BooleanEditor;
    UrlPreview: StringEditor;
    UploadPath: StringEditor;
}

export class StorageForm extends PrefixedContext {
    static readonly formKey = 'Settings.Storage';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!StorageForm.init)  {
            StorageForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = BooleanEditor;

            initFormType(StorageForm, [
                'AccessKeyId', w0,
                'SecretAccessKey', w0,
                'Endpoint', w0,
                'Region', w0,
                'BucketName', w0,
                'S3Types', w1,
                'IsDefault', w2,
                'UrlPreview', w0,
                'UploadPath', w0
            ]);
        }
    }
}

[EStorageType]; // referenced types

File: GXpert/GXpert.Web/Modules/Settings/Storage/StorageForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Settings.Forms;

[FormScript("Settings.Storage")]
[BasedOnRow(typeof(StorageRow), CheckNames = true)]
public class StorageForm
{
    public string AccessKeyId { get; set; }
    public string SecretAccessKey { get; set; }
    public string Endpoint { get; set; }
    public string Region { get; set; }
    public string BucketName { get; set; }
    public short S3Types { get; set; }
    public bool IsDefault { get; set; }
    public string UrlPreview { get; set; }
    public string UploadPath { get; set; }
}